src/utils/ota/restart.go

```diff
@@ -0,0 +1,73 @@
+package ota
+
+import (
+	"log"
+	"os"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+func Restart(extraArgs ...string) error {
+	executable, err := os.Executable()
+	if err != nil {
+		return errors.Wrap(err, "failed to resolve the path to the current executable")
+	}
+
+	workingDirectory, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "failed to resolve the current working directory")
+	}
+
+	execSpec := &syscall.ProcAttr{
+		Dir: workingDirectory,
+		Env: os.Environ(),
+		Files: []uintptr{
+			os.Stdin.Fd(),
+			os.Stdout.Fd(),
+			os.Stderr.Fd(),
+		},
+	}
+
+	var args []string
+
+	if len(extraArgs) != 0 {
+		args = mergeExtraArgs(os.Args, extraArgs)
+	} else {
+		args = os.Args
+	}
+
+	fork, err := syscall.ForkExec(executable, args, execSpec)
+	if err != nil {
+		return errors.Wrap(err, "failed to spawn a new process")
+	}
+
+	log.Printf("new process has been started successfully [old_pid=%d,new_pid=%d]\n",
+		os.Getpid(), fork)
+
+	os.Exit(0)
+	return nil
+}
+
+func mergeExtraArgs(osArgs, extraArgs []string) []string {
```

Comment:
This function is unnecessary complex and doesn't save us from errors: what if extraArgs is something like ["--pprof",":8081"] and os args already include --pprof :8080 ?

Reply:
I would be careful not accepting duplicated args here, because there are programs accepting multiple flags. For example - docker :) You can add multiple ports with the `-p` flag: `-p 8080:80/tcp -p 8080:80/udp`

Reply:
The only purpose of passing additional arguments to the next process is allow it to skip the version check at the startup with the `-skip-update-check-on-start`. It's not necessary, but I was aware of the situation where there is still no version is being embedded into the binary file in the release, so this way it would be an endless update loop.

~~~
(1) Binary version v0.0.1 (default) -> (2) Finds a version v0.7.0 (no version is embedded to it) -> (3) v0.0.1 Shuts down to run the v0.7.0 -> (4) v0.7.0 starts without knowing what version it is -> (5) Finds v0.7.0 -> Go to step (3)
~~~

This flag allows to defer this situation, and with a delay 24 hours it would still work without the embedded version and still downloading the latest version

Reply:
I'm ok with it checking for duplicates, it's just that it should be done correctly. Also considering how current flags implementation works I'd rather move this kind of fix into another PR since other parts of it are quite straightforward but with this one we could explore multiple options and I don't want this process to prevent other changes from being merged

Reply:
Yes, I do completely agree with you. It's much more complicated process to properly merge additional arguments, but for the given use-case, I believe, it should be fine not to duplicate a simple flag